aws-rds-optiongroup/aws-rds-optiongroup.json

```diff
@@ -0,0 +1,204 @@
+    },
+    "Tag": {
+      "description": "A key-value pair to associate with a resource.",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "Key": {
+          "type": "string",
+          "description": "The key name of the tag. You can specify a value that is 1 to 127 Unicode characters in length and cannot be prefixed with aws:. You can use any of the following characters: the set of Unicode letters, digits, whitespace, _, ., /, =, +, and -. ",
+          "minLength": 1,
+          "maxLength": 127
+        },
+        "Value": {
+          "type": "string",
+          "description": "The value for the tag. You can specify a value that is 1 to 255 Unicode characters in length and cannot be prefixed with aws:. You can use any of the following characters: the set of Unicode letters, digits, whitespace, _, ., /, =, +, and -. ",
+          "minLength": 1,
+          "maxLength": 255
+        }
+      },
+      "required": [
+        "Key",
+        "Value"
+      ]
+    }
+  },
+  "properties": {
+    "OptionGroupName": {
+      "description": "Specifies the name of the option group.",
+      "type": "string",
+      "pattern": "^[a-zA-Z]{1}(?:-?[a-zA-Z0-9]){0,254}$"
+    },
+    "OptionGroupDescription": {
+      "description": "Provides a description of the option group.",
+      "type": "string"
+    },
+    "EngineName": {
+      "description": "Indicates the name of the engine that this option group can be applied to.",
+      "type": "string",
+      "enum": [
+        "aurora",
+        "aurora-mysql",
+        "aurora-postgresql",
+        "mariadb",
+        "mysql",
+        "oracle-ee",
+        "oracle-se2",
+        "oracle-se1",
+        "oracle-se",
+        "postgres",
+        "postgresql-license",
+        "sqlserver-ee",
+        "sqlserver-se",
+        "sqlserver-ex",
+        "sqlserver-web"
+      ]
+    },
+    "MajorEngineVersion": {
+      "description": "Indicates the major engine version associated with this option group.",
+      "type": "string"
+    },
+    "OptionConfigurations": {
+      "description": "Indicates what options are available in the option group.",
+      "type": "array",
+      "uniqueItems": true,
+      "insertionOrder": false,
+      "items": {
+        "$ref": "#/definitions/OptionConfiguration"
+      }
+    },
+    "Tags": {
+      "type": "array",
+      "uniqueItems": true,
+      "insertionOrder": false,
+      "description": "An array of key-value pairs to apply to this resource.",
+      "items": {
+        "$ref": "#/definitions/Tag"
+      }
+    }
+  },
+  "additionalProperties": false,
+  "required": [
+    "EngineName",
+    "MajorEngineVersion",
+    "OptionGroupDescription"
+  ],
+  "createOnlyProperties": [
+    "/properties/EngineName",
+    "/properties/MajorEngineVersion",
+    "/properties/OptionGroupDescription",
+    "/properties/OptionGroupName"
+  ],
+  "primaryIdentifier": [
+    "/properties/OptionGroupName"
```

Comment:
primary identifier must be either create or read only. I dont believe we allow customers to provide value for this property.

Reply:
Made it read-only. Thanks!